optics-core/src/Optics/At/Core.hs

```diff
@@ -115,7 +118,13 @@ instance Ord a => Contains (Set a) where
 -- | Type family that takes a key-value container type and returns the type of
 -- values stored in the container, for example @'IxValue' ('Map' k a) ~ a@. This
 -- is shared by both 'Ixed' and 'At'.
-type family IxValue (m :: *) :: *
+type family IxValue (m :: Type) :: Type
+
+-- | Type family that takes a key-value container type and returns the kind
+-- of optic to index into it. For most containers, it's 'An_AffineTraversal',
+-- @Representable@ (Naperian) containers it is 'A_Lens', and multi-maps would
+-- have 'A_Traversal'.
+type family IxKind (m :: Type) :: OpticKind
```

Comment:
Would it make sense to have `IxKind` be an associated type in `Ixed`, rather than a top-level type family? Then it could have a default definition of `An_AffineTraversal`, which would save a few LOC and be more backwards/`lens`-compatible.